### ISSUE TITLE
Nordigen import not working correctly in Data Importer. Some transactions are said to have the same source and destination

### ISSUE THREAD

**Issue by Ugdalfson** [firefly-iii/firefly-iii#5345]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

When I import with Nordigen via Data Importer multiple transactions are not added to the asset account and I get the error:
```
    (2)transactions.0.destination_id: Source and destination are the same. (original value: "49")
    (2)transactions.0.destination_name: Source and destination are the same. (original value: "(not found)")
```

I tried to create another account which I set as default if some account could not be determined but it did not help.

### Expected behaviour

The example given in the debug information section should be added to the transactions.

### Debug information

For an example transaction in json format:
```
            {
                "bankTransactionCode": "PMNT",
                "bookingDate": "2010-01-01",
                "creditorAccount": {
                    "iban": "XY98765432101234567"
                },
                "creditorName": "Erika Mustermann",
                "debtorAccount": {
                    "iban": "XY12345678910111213"
                },
                "debtorName": "Max Mustermann",
                "endToEndId": "NOTPROVIDED",
                "proprietaryBankTransactionCode": "CODEBLABLA",
                "purposeCode": "RINP",
                "remittanceInformationUnstructured": "SomeText",
                "remittanceInformationUnstructuredArray": [
                    "SomeText"
                ],
                "transactionAmount": {
                    "amount": "80.00",
                    "currency": "EUR"
                },
                "valueDate": "2010-01-01"
            },
```

I get the following error in the data importer:
```
[2021-12-02 17:30:41] production.DEBUG: Submitting to Firefly III: {"apply_rules":true,"error_if_duplicate_hash":false,"transactions":[{"type":"transfer","date":"2010-01-01","amount":"80.00","description":"SomeText","order":0,"currency_code":"EUR","tags":[],"category_name":null,"category_id":null,"external_id":"blabla","internal_reference":"","destination_id":52,"source_id":52}]} 

[2021-12-02 17:30:41] production.ERROR: Submission error: 0 ["Source and destination are the same."] 

[2021-12-02 17:30:41] production.DEBUG: Add error on index #100 (line no. 101): transactions.0.destination_id: Source and destination are the same. (original value: "52")  
```

It seems like the importer does not have enough information to distinguish the two accounts, although the two IBANS should be enough, right?

**Comment by JC5** [comment]
The 2 IBAN's could be mixed up, if there are settings somewhere that maps both IBANs to the same destination.

Your debug logs should have something like this, during the conversion step:

```
[IBAN] as a Firefly III asset account so this is a transfer
```

If this is not the case, the importer would have submitted the IBAN's and not the ID's.

**Comment by Ugdalfson** [comment]
Thanks for the quick answer! I still do not know how to fix this. I did not change any settings, so it should be the default installation mostly. I can not find this line in the debug logs. What I just noticed is that Expense accounts are generated as expected. However, the importer did not create any revenue accounts. If both asset accounts exist, the transfer is created correctly. If the transfer goes from my account to another account it is also created correctly (expense account created automatically). If the transfer goes from some unknown account to my account the transfer is not created (no revenue account created) and I get the error message above.
